htdocs/core/class/html.formfile.class.php

```diff
@@ -1320,5 +1328,34 @@ public function listOfLinks($object, $permtodelete=1, $action=null, $selected=nu
         return $nboflinks;
     }
 
+    /**
+     * Show detail icon with link for preview
+     * @param array     $file   File
+     * @param string    $modulepart     propal, facture, facture_fourn, ...
+     * @param string    $relativepath   Relative path of docs
+     * @return string   $out            Output string with HTML
+     */
+    function showPreview($file, $modulepart, $relativepath){
+        global $langs;
+
+        $out='';
+
+        $mime_preview = array('jpeg', 'png', 'gif', 'tiff', 'pdf', 'plain', 'css');
+        $num_mime = array_search(dol_mimetype($file['name'], '', 1), $mime_preview);
+
+        if( $num_mime!== false){
+            $out.= ' <a href="javascript:document_preview(\''.DOL_URL_ROOT.'/document.php?modulepart='.$modulepart.'&amp;attachment=0&amp;file='.urlencode($relativepath).'\',\''.dol_mimetype($file['name']).'\', \''.$file['name'].'\'';
+            if($num_mime<=3){
+                $size = getimagesize($file[fullname]);
```

Comment:
What happened if image is vary larger than screen ?
Why providing image size here. The popup for a preview should be not larger than 80% of user screen.
You can use $_SESSION["dol_screenwidth"] and $_SESSION["dol_screenheight"] to get value of visible screen (they are calculated and defined at user login).